### ISSUE TITLE
Vim9: "here" written on terminal when using nested closure

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#7653]
**Describe the bug**

In Vim9 script, using a nested closure might cause Vim to write `here` on the terminal.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -i NONE -U NONE -S <(cat <<'EOF'
        vim9
        def FuncB(l: any)
            map(l, (_, v) => map(l, (_, v) => v))
        enddef
        def FuncA()
            FuncB([])
        enddef
        FuncA()
        q
    EOF
    )

The text `here` is written on the terminal after Vim is quit.

**Expected behavior**

Nothing is written on the terminal after Vim is quit.

**Screenshots**

![gif](https://user-images.githubusercontent.com/8505073/104133122-bfe0cf00-5381-11eb-87a8-01899d585fe1.gif)

**Environment**

 - Vim version: 8.2 Included patches: 1-2323
 - OS: Ubuntu 16.04.7 LTS
 - Terminal: xterm(363)

**Additional context**

Re-writing `FuncA()` into a legacy function fixes the issue:
```vim
vim9
def FuncB(l: any)
    map(l, (i, v) => map(l, (_, v) => v)[i])
enddef
fu FuncA()
    call FuncB([])
endfu
FuncA()
q
```
---

Regression introduced in [8.2.2322](https://github.com/vim/vim/releases/tag/v8.2.2322).

**Comment by lacygoill** [comment]
It's not just visible after quitting Vim; I can see multiple lines of `hereherehere...` while Vim is running, either on the command-line or in a Vim window.  They can be erased with `:redraw`.  I'm going to stay on 8.2.2321 for the moment.

**Comment by brammool** [comment]
Fixed by 8.2.2327
